src/lakefs_spec/spec.py

```diff
@@ -618,13 +619,15 @@ def __init__(
         )
 
         global _warn_on_fileupload
-        if mode == "wb" and _warn_on_fileupload:
-            logger.warning(
-                f"Calling `{self.__class__.__name__}.open()` in write mode results in unbuffered "
-                "file uploads, because the lakeFS Python client does not support multipart "
-                "uploads. Uploading large files unbuffered can have performance implications."
-            )
-            _warn_on_fileupload = False
+        if mode == "wb":
+            if _warn_on_fileupload:
```

Comment:
Let's move the `global _warn_on_fileupload` into this `if` statement, since it is only relevant there.